addons/l10n_in_edi/models/account_move.py

```diff
@@ -1,57 +1,694 @@
+        if self.l10n_in_edi_attachment_id:
+            return json.loads(self.l10n_in_edi_attachment_id.sudo().raw.decode("utf-8"))
+
+    def _l10n_in_lock_invoice(self):
+        try:
+            with self.env.cr.savepoint(flush=False):
+                self.env.cr.execute(
+                    "SELECT * FROM account_move WHERE id IN %s FOR UPDATE NOWAIT",
+                    [tuple(self.ids)]
+                )
+        except psycopg2.errors.LockNotAvailable:
+            raise UserError(_('This electronic document is being processed already.')) from None
+
+    def _l10n_in_edi_send_invoice(self):
+        self.ensure_one()
+        if self.l10n_in_edi_error:
+            # make sure to clear the error before sending again
+            self.l10n_in_edi_error = False
+        partners = set(self._get_l10n_in_seller_buyer_party().values())
+        for partner in partners:
+            if partner_validation := partner._l10n_in_edi_strict_error_validation():
+                return partner_validation
+        self._l10n_in_lock_invoice()
+        generate_json = self._l10n_in_edi_generate_invoice_json()
+        response = self._l10n_in_edi_connect_to_server(
+            url_end_point='generate',
+            json_payload=generate_json
+        )
+        if error := response.get('error', {}):
+            odoobot_id = self.env['ir.model.data']._xmlid_to_res_id('base.partner_root')
+            error_codes = [e.get("code") for e in error]
+            if '2150' in error_codes:
+                # Get IRN by details in case of IRN is already generated
+                # this happens when timeout from the Government portal but IRN is generated
+                response = self._l10n_in_edi_connect_to_server(
+                    url_end_point='getirnbydocdetails',
+                    params={
+                        "doc_type": (
+                            (self.move_type == "out_refund" and "CRN")
+                            or (self.debit_origin_id and "DBN")
+                            or "INV"
+                        ),
+                        "doc_num": self.name,
+                        "doc_date": self.invoice_date and self.invoice_date.strftime("%d/%m/%Y"),
+                    }
+                )
+                if not response.get("error"):
+                    error = []
+                    link = Markup(
+                        "<a href='https://einvoice1.gst.gov.in/Others/VSignedInvoice'>%s</a>"
+                    ) % (_("here"))
+                    self.message_post(
+                        author_id=odoobot_id,
+                        body=_(
+                            "Somehow this invoice has been submited to government before."
+                            "%(br)sNormally, this should not happen too often"
+                            "%(br)sJust verify value of invoice by upload json to government website %(link)s.",
+                            br=Markup("<br/>"),
+                            link=link
```

Comment:
```suggestion
                        body=Markup("%s <br/> %s <br/>") % (_("Somehow....)
```
Like the link above...

The problem in how it is now, is that it generates a string, not a Markup, so the Markup is  useless.

Reply:
ok, so the translation makes sure it stays Markup